### ISSUE TITLE
[Bug]: OpenVINO vs Torch inferencer on CPU gives significantly different results

### ISSUE THREAD

**Issue by FedericoDeBona** [openvinotoolkit/anomalib#2448]
### Describe the bug

From #2447, training a model, then exporting it in both OpenVINO and Torch the visual result and pred_score are a lot different. Looks like a normalization problem but in the metadata of both models, the values are the same.

**OpenVINO inference** - pred_score: 0.7632662161599639
![openvino](https://github.com/user-attachments/assets/2128917a-678a-4851-8325-447d969dc00a)
**Torch inference** - pred_score: 0.6897885799407959
![torch](https://github.com/user-attachments/assets/1c68b5e0-2b96-4840-96b7-a4d8ad73bd00)

## Full code
```python
from anomalib.data import MVTec
from anomalib.models import Patchcore
from anomalib.engine import Engine
from anomalib.deploy import ExportType
from anomalib.deploy import OpenVINOInferencer, TorchInferencer
from PIL import Image
from glob import glob
from anomalib.utils.visualization import ImageVisualizer
from anomalib.utils.visualization.image import  VisualizationMode
from anomalib import TaskType
import os
import cv2

CATEGORY="screw"

datamodule = MVTec(category=CATEGORY)
model = Patchcore()
engine = Engine()

engine.fit(datamodule=datamodule, model=model)
engine.export(model=model,export_type=ExportType.TORCH) 
engine.export(model=model,export_type=ExportType.OPENVINO) 

vino_inferencer = OpenVINOInferencer(
    path=f"/home/trainer/trainer_engine/results/Patchcore/MVTec/{CATEGORY}/latest/weights/openvino/model.bin",
    metadata=f"/home/trainer/trainer_engine/results/Patchcore/MVTec/{CATEGORY}/latest/weights/openvino/metadata.json",
    device="CPU")
torch_inferencer = TorchInferencer(path=f"/home/trainer/trainer_engine/results/Patchcore/MVTec/{CATEGORY}/latest/weights/torch/model.pt", device="cpu")
print(vino_inferencer.metadata)
print(torch_inferencer.metadata)
#Output
"""
{'task': 'segmentation', 'image_threshold': 30.5954647064209, 'pixel_threshold': 34.377254486083984, 'anomaly_maps.min': 9.125215530395508, 'anomaly_maps.max': 49.421302795410156, 'pred_scores.min': 25.861536026000977, 'pred_scores.max': 52.29983139038086}
{'task': <TaskType.SEGMENTATION: 'segmentation'>, 'image_threshold': 30.5954647064209, 'pixel_threshold': 34.377254486083984, 'anomaly_maps.min': 9.125215530395508, 'anomaly_maps.max': 49.421302795410156, 'pred_scores.min': 25.861536026000977, 'pred_scores.max': 52.29983139038086}

"""


visualizer = ImageVisualizer(mode=VisualizationMode.FULL, task=TaskType.SEGMENTATION)

for defect_path in sorted(glob(f"/home/trainer/trainer_engine/datasets/{CATEGORY}/test/*")):
	defect = os.path.basename(defect_path)
	print(f"===== {defect} =====")
	for img_path in sorted(glob(f"/home/trainer/trainer_engine/datasets/{CATEGORY}/test/{defect}/*")):
		img_name = os.path.basename(img_path)
		vino_res = vino_inferencer(cv2.imread(img_path))
		torch_res = torch_inferencer(img_path)
		print("VINO")
		print(vino_res.pred_score)
		display(Image.fromarray(cv2.resize(visualizer.visualize_image(vino_res), (500*3,125*3))))
		print("TORCH")
		print(torch_res.pred_score)
		display(Image.fromarray(cv2.resize(visualizer.visualize_image(torch_res), (500*3,125*3))))
```

### Dataset

MVTec

### Model

PatchCore

### Steps to reproduce the behavior

See above

### OS information

OS information:
- OS: Ubuntu 24.04
- Python version: 3.10.14
- Anomalib version: 1.2.0.dev0
- PyTorch version: 2.4.0+cu118
- CUDA/cuDNN version: 11.8
- GPU models and configuration: GeForce RTX 3090 Ti
- Any other relevant information: I'm using a custom dataset (link above)

### Expected behavior

Not sure, I suppose the exported models should have almost equal result in both OpenVINO and Torch

### Pip/GitHub

GitHub

### What version/branch did you use?

1.2.0.dev0

### Configuration YAML

```yaml
-
```

### Logs

```shell
-
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by pdxmusic** [comment]
I'm also facing this kind of issue, did you find any solution yet?

**Comment by FedericoDeBona** [comment]
https://github.com/openvinotoolkit/anomalib/issues/2559#issuecomment-2662553308
